File: cpp/1323.maximum-69-number.cpp
#include <cmath>

class Solution {
public:
  int maximum69Number(int num) {
    int pos = -1;

    for (int x = num, i = 0; x > 0; x /= 10, i++) {
      if (x % 10 == 6) {
        pos = i; // don't break on first match, go on to find most significant
                 // digit 6
      }
    }

    if (pos == -1) {
      return num;
    }

    return num + (3 * std::pow(10, pos));
  }
};

File: golang/1323.maximum-69-number.go
func maximum69Number(num int) int {
	pos := -1

	for x, i := num, 0; x > 0; x, i = x/10, i+1 {
		if x%10 == 6 {
			pos = i
		}
	}

	if pos == -1 {
		return num
	}

	p := 3
	for range pos {
		p *= 10
	}

	return num + p
}
